File: backend/src/main/java/by/minilooth/diploma/service/stores/StoreService.java
package by.minilooth.diploma.service.stores;

import by.minilooth.diploma.exception.ActionIsImpossibleException;
import by.minilooth.diploma.exception.stores.StoreNotFoundException;
import by.minilooth.diploma.models.bean.stores.Address;
import by.minilooth.diploma.models.bean.stores.Store;
import by.minilooth.diploma.models.stores.ProcessStore;
import by.minilooth.diploma.models.stores.StoreFilter;
import by.minilooth.diploma.models.stores.StoreList;

import java.util.List;
import java.util.Optional;

public interface StoreService {

    Long ITEMS_PER_PAGE = 9L;

    void save(Store store);
    Store save(ProcessStore processStore);
    Store update(ProcessStore processStore, Long id) throws StoreNotFoundException;
    void delete(Store store);
    Store delete(Long id) throws StoreNotFoundException;
    Store getById(Long id) throws StoreNotFoundException;
    StoreList getAll(StoreFilter storeFilter);
    List<Store> getAll();
    Boolean existsByAddress(Address address);

}

File: frontend/service/stores/StoreService.ts
import {IncomingHttpHeaders} from "http";
import {ProcessStore, Store, StoreFilter, StoreList} from "types/stores/store";
import {Axios} from "core/axios";

export class StoreService {

  private static readonly URL = '/store';

  static async getAll(filter: StoreFilter, headers?: IncomingHttpHeaders): Promise<StoreList> {
    const {data: stores} = await Axios.post<StoreList>(this.URL, filter, {headers});
    return stores;
  }

  static async add(processStore: ProcessStore): Promise<Store> {
    const {data: store} = await Axios.put<Store>(this.URL, processStore);
    return store;
  }

  static async update(id: number | string, processStore: ProcessStore): Promise<Store> {
    const {data: store} = await Axios.patch<Store>(`${this.URL}/${id}`, processStore);
    return store;
  }

  static async delete(id: number | string): Promise<Store> {
    const {data: store} = await Axios.delete<Store>(`${this.URL}/${id}`);
    return store;
  }

}
